app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/index.tsx

```diff
@@ -0,0 +1,139 @@
+import T from 'i18n-react';
+import React, { ChangeEvent, useEffect } from 'react';
+import styled from 'styled-components';
+
+const FormFieldWrapper = styled(Box)`
+  width: calc(100% - 60px);
+  margin-right: 60px;
+  margin-bottom: 15px;
+`;
+
+const CheckBox = styled(InputCheckbox)`
+  display: flex;
+  width: 100%;
+  margin-bottom: 0px;
+`;
+
+const Label = styled(Typography)`
+  font-style: normal;
+  font-weight: 400;
+  font-size: 14px;
+  line-height: 150%;
+  letter-spacing: 0.15;
+  color: #5f6368;
+`;
+
+const SelectField = styled(InputSelect)`
+  height: 40px;
+  background: #FFFFFF;
+  border: 1px solid #DADCE0;
+  border-radius: 4,
+  padding: 5px 15px 0px 15px;
+  font-size: 14px;
+  & :before: {
+  display: none;
+  }
+  & :focus-visible: {
+  outline: none !important;
+  }
+  & :after: {
+  display: none;
+  }
+  & .MuiInputBase-input': {
+  padding: 6px 0px 11px;
+  }
+`;
+
+export default function({ values, changeEventListener }: IParsingPopupBodyProps) {
+  const classes = useStyles();
+
+  const { format, fileEncoding, enableQuotedValues, skipHeader } = values;
+  let selectedFormatValue: IOptions[] = [];
+  let selectedEncodingValue: IOptions[] = [];
+
+  useEffect(() => {
+    selectedFormatValue = FORMAT_OPTIONS?.filter((i) => i.value === format);
+  }, [format]);
+
+  useEffect(() => {
+    selectedEncodingValue = CHAR_ENCODING_OPTIONS?.filter((i) => i.value === fileEncoding);
+  }, [fileEncoding]);
+
+  return (
+    <Box>
+      <FormFieldWrapper>
+        <Label data-testid="popup-body-label-text-format">
+          {T.translate('features.WranglerNewUI.ParsingDrawer.format')}
+        </Label>
+        <InputSelect
+          classes={{ icon: classes.selectIconStyles, select: classes.selectStyles }}
+          className={classes.selectFieldStyles}
+          optionClassName={{ root: classes.optionStyles }}
+          fullWidth
+          defaultValue={FORMAT_OPTIONS[0].value}
+          value={selectedFormatValue[0]?.value}
+          onChange={(event: ChangeEvent<HTMLSelectElement>) =>
+            changeEventListener(event.target.value, 'format')
+          }
+          options={FORMAT_OPTIONS}
+        />
+      </FormFieldWrapper>
+      <FormFieldWrapper>
+        <Label data-testid="popup-body-label-text-encoding">
+          {T.translate('features.WranglerNewUI.ParsingDrawer.encoding')}
+        </Label>
+        <InputSelect
+          classes={{ icon: classes.selectIconStyles, select: classes.selectStyles }}
+          className={classes.selectFieldStyles}
+          optionClassName={{ root: classes.optionStyles }}
+          defaultValue={CHAR_ENCODING_OPTIONS[0].value}
+          fullWidth
+          value={selectedEncodingValue[0]?.value}
+          onChange={(event: ChangeEvent<HTMLSelectElement>) =>
+            changeEventListener(event.target.value, 'fileEncoding')
+          }
+          options={CHAR_ENCODING_OPTIONS}
+        />
+      </FormFieldWrapper>
+      <CheckBox
```

Comment:
These fields should not be hardcoded. Please look at how the current parsing dialog handles different file types.